utils/generic.py

```diff
@@ -878,4 +932,152 @@ def assign_population_pcs(
+        )
+
+    metrics = set(stats[0])
+    for stat_expr in stats[1:]:
+        metrics = metrics.intersection(set(stat_expr))
+
+    if 'stdev' in metrics:
+        missing_fields = [x for x in ['n', 'mean'] if x not in metrics]
+        if missing_fields:
+            logger.warn(f'Cannot merge `stdev` from give stats counters since they are missing the following fields: {",".join(missing_fields)}')
+            metrics.remove('stdev')
+
+    all_stats = stats.map(lambda x: hl.struct(
+        min=x.min if 'min' in metrics else hl.null(hl.tfloat64),
+        max=x.max if 'max' in metrics else hl.null(hl.tfloat64),
+        mean=x.mean if 'mean' in metrics else hl.null(hl.tfloat64),
+        variance=x.stdev* x.stdev if 'stdev' in metrics else hl.null(hl.tfloat64),
+        n=x.n if 'n' in metrics else hl.null(hl.tfloat64),
+        sum=x.sum if 'sum' in metrics else hl.null(hl.tfloat64)
+    ))
+    agg_stats = all_stats[1:].fold(add_stats, all_stats[0])
+    return agg_stats.select(
+        **{metric: agg_stats[metric] if metric != 'stdev' else hl.sqrt(agg_stats.variance) for metric in metrics}
+    )
+
+
+def merge_sample_qc_expr(sample_qc_exprs: List[hl.expr.StructExpression]) -> hl.expr.StructExpression:
+    """
+    Creates an expression that merges results from non-overlapping strata of hail.sample_qc
+    E.g.:
+    - Compute autosomes and sex chromosomes metrics separately, then merge results
+    - Compute bi-allelic and multi-allelic metrics separately, then merge results
+
+    :param list of StructExpression sample_qc_exprs: Expression to the root of the sample QC struct for eeach stratification
+    :return: Combined sample QC results
+    :rtype: StructExpression
+    """
+    additive_metrics = [
+        'n_called', 'n_not_called', 'n_hom_ref', 'n_het', 'n_hom_var', 'n_snp',
+        'n_insertion', 'n_deletion', 'n_singleton', 'n_transition', 'n_transversion', 'n_star'
+    ]
+    ratio_metrics = [
+                       ('call_rate', 'n_called', 'n_not_called'),
+                       ('r_ti_tv', 'n_transition', 'n_transversion'),
+                       ('r_het_hom_var', 'n_het', 'n_hom_var'),
+                       ('r_insertion_deletion', 'n_insertion', 'n_deletion')
+                   ]
+    stats_metrics = ['gq_stats', 'dp_stats']
+    sample_qc_fields = set(sample_qc_exprs[0])
+    for sample_qc_expr in sample_qc_exprs[1:]:
+        sample_qc_fields = sample_qc_fields.union(set(sample_qc_expr))
+
+    merged_exprs = {
+        metric: hl.sum([sample_qc_expr[metric] for sample_qc_expr in sample_qc_exprs])
+        for metric in additive_metrics if metric in sample_qc_fields
+    }
+
+    merged_exprs.update({
+        metric: hl.float64(divide_null(merged_exprs[nom],merged_exprs[denom]))
+        for metric, nom, denom in ratio_metrics if nom in sample_qc_fields and denom in sample_qc_fields
+    })
+
+    # Use n_called as n for DP and GQ stats -- should work really well unless there is something odd
```

Comment:
I'm moving this to the function main docstring and will expand there --  this was clearly meant for myself :)
```
Note regarding the merging of ``dp_stats`` and ``gq_stats``:
    Because ``n`` is needed to aggregate ``stdev``, ``n_called`` is used for this purpose.
    This should work very well on a standard GATK VCF and it essentially assumes that:
    - samples that are called have `DP` and `GQ` fields
    - samples that are not called do not have `DP` and `GQ` fields
    Even if these assumptions are broken for some genotypes, it shouldn't matter too much.
```